atc/db/base_resource_type.go

```diff
@@ -45,14 +49,20 @@ func (brt BaseResourceType) FindOrCreate(tx Tx, unique bool) (*UsedBaseResourceT
 
 func (brt BaseResourceType) Find(runner sq.Runner) (*UsedBaseResourceType, bool, error) {
 	var id int
+	var name string
 	var unique bool
-	err := psql.Select("id, unique_version_history").
-		From("base_resource_types").
-		Where(sq.Eq{"name": brt.Name}).
-		Suffix("FOR SHARE").
+	var defaultsString sql.NullString
+	sb := psql.Select("id, name, unique_version_history, defaults").
+		From("base_resource_types")
+	if brt.Id > 0 {
+		sb = sb.Where(sq.Eq{"id": brt.Id})
+	} else {
```

Comment:
As far as I can tell, this table always starts index at 1. What prompted this change?
https://github.com/concourse/concourse/blob/64a12a7023f7a68993730c3add624e38773eb9fd/atc/db/migration/migrations/1510262030_initial_schema.up.sql#L72-L77

Reply:
With the second commit, this is no longer an issue.